vscode-client/snippets/snippets.json

```diff
@@ -0,0 +1,214 @@
+        "body": [
+            "#!/usr/bin/env ${1|bash,sh|}"
+        ]
+    },
+    "if": {
+        "description": "[if]",
+        "prefix": "if",
+        "body": [
+            "if ${1:command}; then",
+            "\t$0",
+            "fi"
+        ]
+    },
+    "ife": {
+        "description": "[if] [e]lse",
+        "prefix": "ife",
+        "body": [
+            "if ${1:command}; then",
+            "\t${2:echo}",
+            "else",
+            "\t$0",
+            "fi"
+        ]
+    },
+    "wh": {
+        "description": "[wh]ile",
+        "prefix": "wh",
+        "body": [
+            "while ${1:command}; do",
+            "\t$0",
+            "done"
+        ]
+    },
+    "un": {
+        "description": "[un]til",
+        "prefix": "un",
+        "body": [
+            "until ${1:command}; do",
+            "\t$0",
+            "done"
+        ]
+    },
+    "fo": {
+        "description": "[fo]r",
+        "prefix": "fo",
+        "body": [
+            "for ${1:variable} in ${2:list}; do",
+            "\t$0",
+            "done"
+        ]
+    },
+    "fu": {
+        "description": "[fu]nction",
+        "prefix": "fu",
+        "body": [
+            "${1:function_name}() {",
+            "\t$0",
+            "}"
+        ]
+    },
+    "ma": {
+        "description": "[ma]in",
+        "prefix": "ma",
+        "body": [
+            "main() {",
+            "\t$0",
+            "}"
+        ]
+    },
+    ":-": {
+        "description": "[:-] expansion",
+        "prefix": ":-",
+        "body": [
+            "\"\\${${1:variable}:-${2:default}}\""
+        ]
+    },
+    ":=": {
+        "description": "[:=] expansion",
+        "prefix": ":=",
+        "body": [
+            "\"\\${${1:variable}:=${2:default}}\""
+        ]
+    },
+    ":?": {
+        "description": "[:?] expansion",
+        "prefix": ":?",
+        "body": [
+            "\"\\${${1:variable}:?${2:error_message}}\""
+        ]
+    },
+    ":+": {
+        "description": "[:+] expansion",
+        "prefix": ":+",
+        "body": [
+            "\"\\${${1:variable}:+${2:alternative}}\""
+        ]
+    },
+    "#": {
+        "description": "[#] expansion",
+        "prefix": "#",
+        "body": [
+            "\"\\${${1:variable}#${2:pattern}}\""
+        ]
+    },
+    "##": {
+        "description": "[##] expansion",
+        "prefix": "##",
+        "body": [
+            "\"\\${${1:variable}##${2:pattern}}\""
+        ]
+    },
+    "%": {
+        "description": "[%] expansion",
+        "prefix": "%",
+        "body": [
+            "\"\\${${1:variable}%${2:pattern}}\""
+        ]
+    },
+    "%%": {
+        "description": "[%%] expansion",
+        "prefix": "%%",
+        "body": [
+            "\"\\${${1:variable}%%${2:pattern}}\""
+        ]
+    },
+    "..": {
+        "description": "brace expansion",
+        "prefix": "..",
+        "body": [
+            "{${1:from}..${2:to}}"
+        ]
+    },
+    "echo": {
```

Comment:
I'm not sure how much value very basic snippets like `echo` and `source` provides.

Reply:
Just to save some keystrokes and quote argument automatically. :)